pygdf/dataframe.py

```diff
@@ -173,6 +182,11 @@ def __getitem__(self, arg):
             for k, col in self._cols.items():
                 df[k] = col[arg]
             return df
+        elif isinstance(arg, (list,)):
```

Comment:
Should this be `isinstance(arg, collections.abc.Sequence)`, so that this applies to more than just lists, such as tuples.

Reply:
It would also be great if this case had tests.

Reply:
I think you're right, this would be better.  I never used anything different than lists for accessing like this in pandas, so I didn't think about it :).

Reply:
haha no worries :)

Reply:
> Should this be isinstance(arg, collections.abc.Sequence), so that this applies to more than just lists, such as tuples.

I think this should just be `list`.  Because, in pandas, `__getitem__` with a tuple means indexing into MultiIndex.  And, `__getitem__` in pandas is overloaded to do too many things in weird fashion.

Reply:
Unless you define that tuple as a column name in which case it then grabs the column as opposed to indexing. Pandas API in this respect is overly complex.

I'm happy to restrict this to just lists for now, but we should open an issue to iron out what we think the API for selecting columns and indexing should be.

Reply:
So for now I'll leave it supporting a list of columns (adding tests in a new commit), and open the discussion of the API if everyone is alright with that. After a bit of playing I'm definitely not a dan of the bogus treatment of tuples in pandas.